### ISSUE TITLE
Intermediate Epoch Validation Metrics

### ISSUE THREAD

**Issue by Gelesh** [keras-team/keras#17697]
Especially when each individual data point is large, such as an image data, the batch size need to be small to fit  within the memory. Hence in every epoch there would be too many batch, and each batch would update weights. It may be possible that there is a the validation loss minimum was some where mid way to an end of epoch, and further, the model was mostly overfitting. 
This can be handled by provisioning an intermediate epoch validation matric, like for every batch_ID % n == 0 , where n is configurable, a validation metric is logged.

**Comment by tilakrayal** [comment]
@Gelesh,
Yeah as mentioned the things are doable,  it will create the confusion for the user and also when we are trying to provide the results, it might take longer time interval to run the model which will leads to perfomance issue.  If you provide any Usecase for the similar feature, it will be easy to analyse the issue. 
We used custom callback for a while until I found that model.validate resets all metrics so each time we do **model.validate** while the epoch is in progress, we lose all training metrics, overwriting them with validation metrics.
https://www.tensorflow.org/guide/keras/customizing_what_happens_in_fit?hl=en
Thank you!
